python/variable.cpp

```diff
@@ -250,38 +253,76 @@ void init_variable(py::module &m) {
-            &rebin),
-        py::call_guard<py::gil_scoped_release>(),
-        "Returns a new Variable whose data is rebinned with new bin edges.");
+        py::call_guard<py::gil_scoped_release>(), R"(
+        Concatenate input variables along the given dimension.
+
+        Concatenation can happen in two ways:
+        - Along an existing dimension, yielding a new dimension extent given by the sum of the input's extents.
+        - Along a new dimension that is not contained in either of the inputs, yielding an output with one extra dimensions.
+
+        :raises: If the dtype or unit does not match, or if the dimensions and shapes are incompatible.
+        :return: New variable containing all elements of the input variables.
+        :rtype: Variable)");
   m.def("filter",
         py::overload_cast<const Variable &, const Variable &>(&filter),
         py::call_guard<py::gil_scoped_release>());
-  m.def("sum", py::overload_cast<const Variable &, const Dim>(&sum),
-        py::call_guard<py::gil_scoped_release>(),
-        "Returns a new Variable containing the sum of the data along the "
-        "specified dimension.");
   m.def("mean", py::overload_cast<const Variable &, const Dim>(&mean),
-        py::call_guard<py::gil_scoped_release>(),
-        "Returns a new Variable containing the mean of the data along the "
-        "specified dimension.");
+        py::call_guard<py::gil_scoped_release>(), R"(
+        The element-wise mean over the specified dimension.
+
+        :raises: If the dtype cannot be summed, e.g., if it is a string
+        :seealso: :py:class:`scipp.sum`
+        :return: New variable containing the mean.
+        :rtype: Variable)");
   m.def("norm", py::overload_cast<const Variable &>(&norm),
+        py::call_guard<py::gil_scoped_release>(), R"(
+        The element-wise norm.
+
+        :raises: If the dtype does not norm, i.e., if it is not a vector
+        :seealso: :py:class:`scipp.abs` for scalar dtype
+        :return: New variable with scalar elements computed as the norm values if the input elements.
+        :rtype: Variable)");
+  m.def("rebin",
+        py::overload_cast<const Variable &, const Variable &, const Variable &>(
+            &rebin),
         py::call_guard<py::gil_scoped_release>(),
-        "Returns a new Variable containing the norm of the data.");
-  // find out why py::overload_cast is not working correctly here
-  m.def("sqrt", [](const Variable &self) { return sqrt(self); },
+        "Returns a new Variable whose data is rebinned with new bin edges.");
+  m.def("split",
+        py::overload_cast<const Variable &, const Dim,
+                          const std::vector<scipp::index> &>(&split),
         py::call_guard<py::gil_scoped_release>(),
-        "Returns a new Variable containing the square root of the data.");
+        "Split a Variable along a given Dimension.");
+  m.def("sqrt", [](const Variable &self) { return sqrt(self); },
```

Comment:
afaik these exports are not tested.

Reply:
Added tests, expect for some cases of more special functions that may still be changed or removed.